app/src/org/commcare/android/session/DevSessionRestorer.java

```diff
@@ -108,26 +108,36 @@ public static void saveSessionToPrefs() {
         if (ccApp == null) {
             return;
         }
-
         SharedPreferences prefs = ccApp.getAppPreferences();
+        String serializedSession = getSerializedSessionString();
+        if (!"".equals(serializedSession)) {
+            prefs.edit().putString(CommCarePreferences.CURRENT_SESSION, serializedSession).commit();
+        }
+    }
+
+    public static String getSerializedSessionString() {
```

Comment:
Ah, I think I was confusing what you're calling the "readable session string" above, and the "Serialized Session String" here.